addons/dragging/functions/fnc_setDraggable.sqf

```diff
@@ -6,20 +6,51 @@
  * Arguments:
  * 0: Object <OBJECT>
  * 1: True to enable dragging, false to disable <BOOL>
- * 2: Position offset for attachTo command (optional; default: [0, 1.5, 0]) <ARRAY>
- * 3: Direction in degrees to rotate the object after attachTo (optional; default: 0) <NUMBER>
+ * 2: Position offset for attachTo command <ARRAY> (default: [0, 1.5, 0])
+ * 3: Direction in degrees to rotate the object after attachTo <NUMBER> (default: 0)
  * 4: Override weight limit <BOOL> (default: false)
+ * 5: Apply globally <BOOL> (default: false)
  *
  * Return Value:
  * None
  *
  * Example:
- * [cursorTarget, true, [0, 0, 0], 0, false] call ace_dragging_fnc_setDraggable;
+ * [cursorTarget, true, [0, 0, 0], 0, false] call ace_dragging_fnc_setDraggable
  *
  * Public: Yes
  */
 
-params ["_object", "_enableDrag", "_position", "_direction", ["_ignoreWeightDrag", false, [false]]];
+params [
+    ["_object", objNull, [objNull]],
+    ["_enableDrag", false, [false]],
+    "_position",
+    "_direction",
+    ["_ignoreWeightDrag", false, [false]],
+    ["_global", false, [false]]
+];
+
+if (isNull _object) exitWith {};
+
+if (!isNil "_position" && {!(_position isEqualType []) || {!(_position isEqualTypeArray [0, 0, 0])}}) exitWith {
+    ERROR_2("setDraggable: Bad position parameter [%1] for [%2], should be a 3D position or nil",_position,_object);
+};
+
+if (!isNil "_direction" && {!(_direction isEqualType 0)}) exitWith {
+    ERROR_2("setDraggable: Bad direction parameter [%1] for [%2], should be a number or nil",_direction,_object);
+};
```

Comment:
What does this do diff from the `params` ?

Reply:
https://github.com/acemod/ACE3/blob/a894fc78c3a1d49ed335634f92fddc578922b615/addons/dragging/functions/fnc_setCarryable.sqf#L24-L30
still needs to be maintained, so can't use `params` here.